pkg/runtime/ruby/ruby.go

```diff
@@ -102,6 +102,122 @@ func IsInterpreter(proc procfs.Proc) (bool, error) {
 	return false, nil
 }
 
+// versionFromSymbol reads the Ruby version from the ELF symbol table.
+func versionFromSymbol(f *os.File) (string, error) {
+	// PERF(javierhonduco): Using Go's ELF reader in the stdlib is very
+	// expensive. Do this in a streaming fashion rather than loading everything
+	// at once.
+	elfFile, err := elf.NewFile(f)
+	if err != nil {
+		return "", fmt.Errorf("error opening ELF: %w", err)
+	}
+
+	symbols, err := elfFile.Symbols()
+	if err != nil {
+		return "", fmt.Errorf("error reading ELF symbols: %w", err)
+	}
+	for _, symbol := range symbols {
+		if symbol.Name == "ruby_version" {
+			rubyVersionBuf := make([]byte, symbol.Size-1)
+			address := symbol.Value
+
+			_, err = f.Seek(int64(address), io.SeekStart)
+			if err != nil {
+				return "", err
+			}
+
+			_, err = f.Read(rubyVersionBuf)
+			if err != nil {
+				return "", err
+			}
+
+			return string(rubyVersionBuf), nil
+		}
+	}
+
+	return "", errors.New("ruby_version symbol not found")
+}
+
+func RuntimeInfo(proc procfs.Proc) (*runtime.Runtime, error) {
+	isRuby, err := IsRuntime(proc)
+	if err != nil {
+		return nil, fmt.Errorf("is runtime: %w", err)
+	}
+	if !isRuby {
+		return nil, nil //nolint:nilnil
+	}
+
+	rt := &runtime.Runtime{
+		Name: "Ruby",
+	}
+
+	exe, err := proc.Executable()
+	if err != nil {
+		return rt, err
+	}
+
+	f, err := os.Open(absolutePath(proc, exe))
+	if err != nil {
+		return rt, fmt.Errorf("open executable: %w", err)
+	}
+	defer f.Close()
+
+	versionString, err := versionFromSymbol(f)
+	if err == nil {
+		version, err := semver.NewVersion(versionString)
+		if err != nil {
+			return rt, fmt.Errorf("new version: %q: %w", versionString, err)
+		}
+		rt.Version = version
+		return rt, nil
+	}
+	if _, err := f.Seek(0, io.SeekStart); err != nil {
```

Comment:
The file is closed in line 163. I think this seek is not needed and will always fail

Reply:
In line 163, it's used with `defer` so the file is open here.
The `Seek` here, because of ```go
			address := symbol.Value

			_, err = f.Seek(int64(address), io.SeekStart)
			if err != nil {
				return "", err
			}

			_, err = f.Read(rubyVersionBuf)
			if err != nil {
				return "", err
			}
``` in `versionFromSymbol`.